squirreldisk: init at 0.3.4

peret opened the pull request:
## Description of changes
Initialize SquirrelDisk, a cross-platform disk usage analyzer written in Rust: https://www.squirreldisk.com/

## Things done
- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

peret committed:
squirreldisk: init at 0.3.4

drupol commented on pkgs/by-name/sq/squirreldisk/package.nix:
```diff
@@ -0,0 +1,87 @@
+
+  src = fetchFromGitHub {
+    owner = "adileo";
+    repo = pname;
```
Please don't use this.

`repo = pname;` is nice for DRY but creates a binding that goes too far.

See further information about this here: https://github.com/nix-community/nixpkgs-lint/issues/21

drupol commented on pkgs/by-name/sq/squirreldisk/package.nix:
```diff
@@ -0,0 +1,87 @@
+    owner = "adileo";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-As2nvc68knjeLPuX0QLBoybj8vuvkpS5Vr+7U7E5CjA=";
```
Prefer `hash = ...` to `sha256 = ...` Both currently work, but the former will make it easier if the algorithm ever needs to change.

Since there could be different hashing algorithms in use, this make sense.

More info: https://nixos.org/manual/nixpkgs/stable/#fetchurl

> Nixpkgs contributors are currently recommended to use hash.

drupol reviewed (changes requested):
Hi !

Thanks for your contribution.

I've left some comments, let me know if this is clear enough.

peret committed:
squirreldisk: fix build for aarch64-linux

The squirreldisk project does not contain a pdu binary for aarch64-linux,
unfortunately. Therefore, use the pdu binary from nixpkgs and copy
it to the project's binary directory _before_ building the Tauri
project

peret commented:
@drupol thank you for the review and for letting me know about these issues :pray:

drupol commented:
The broken flag should stay here ?

peret commented:
> The broken flag should stay here ?

@drupol, sorry, I'm not sure what you are referring to?

drupol commented:
Oops looks like Github on mobile had a brain fart, this comment was supposed to be written for another PR. Please apologize.

peret commented:
> Oops looks like Github on mobile had a brain fart, this comment was supposed to be written for another PR. Please apologize.

No problem! I was getting really confused 😆 Thanks for merging this 🙏
